File: all-submissions/0515-find-largest-value-in-each-tree-row/solution.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<Integer> largestValues(TreeNode root) {
        List<Integer> answer = new ArrayList<>();
        if(root == null) return answer;
        Queue<TreeNode> queue = new LinkedList<>();
        queue.add(root);

        while(!queue.isEmpty()){
            int nodesinCurrLevel = queue.size();
            int maxInCurrLevel = Integer.MIN_VALUE;
            for (int i=0; i<nodesinCurrLevel; i++) {
                TreeNode node = queue.remove();
                maxInCurrLevel = Math.max(maxInCurrLevel, node.val);
                if (node.left != null) {
                    queue.add(node.left);
                }
                if(node.right != null) {
                    queue.add(node.right);
                }
            }
            answer.add(maxInCurrLevel);
        }
        return answer;
    }
}

File: all-submissions/0515-find-largest-value-in-each-tree-row/solution.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def largestValues(self, root: Optional[TreeNode]) -> List[int]:
        if root is None:
            return []
        res = []
        queue = deque()
        queue.append(root)
        while queue:
            nodes_in_curr_level = len(queue)
            currMax = float("-inf")
            for _ in range(nodes_in_curr_level):
                node = queue.popleft()
                currMax = max(currMax, node.val)
                if node.left:
                    queue.append(node.left)
                if node.right:
                    queue.append(node.right)
            res.append(currMax)
        return res
